### PR TITLE
MM-11039

### PR DESCRIPTION
**Pull request by mzaks**
#### Summary
Migrate WebSocket actions to Flow

#### Ticket Link
https://github.com/mattermost/mattermost-server/issues/9762
https://mattermost.atlassian.net/browse/MM-11039

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit tests passed
- [x] Ran `make flow` to ensure type checking passed
- [x] Added or updated unit tests (required for all new features)

#### Test Information
Did not test. Suppose to be a non destructive refactoring.

#### Notes from contributor
I introduce comments in place, which I find need discussion.

### CODE REVIEW

File: src/actions/websocket.js [hunk 1]
```diff
@@ -212,9 +222,10 @@ function handleEvent(msg) {
     case WebsocketEvents.UPDATE_TEAM:
         doDispatch(handleUpdateTeamEvent(msg));
         break;
-    case WebsocketEvents.PATCH_TEAM:
-        doDispatch(handlePatchTeamEvent(msg));
-        break;
+
+    // case WebsocketEvents.PATCH_TEAM:
+    //     doDispatch(handlePatchTeamEvent(msg));
+    //     break;
```
Comment (mzaks) [inline src/actions/websocket.js]: There is no `WebsocketEvents.PATH_TEAM` constant present. I commented out the case. But it is rather needs to be removed. Comments welcome

File: src/actions/websocket.js [hunk 2]
```diff
@@ -355,19 +366,20 @@ function handleUpdateTeamEvent(msg) {
...
-        type: TeamTypes.PATCHED_TEAM,
-        data: JSON.parse(msg.data.team),
-    };
-}
+// function handlePatchTeamEvent(msg) {
+//     return {
+//         type: TeamTypes.PATCHED_TEAM,
+//         data: JSON.parse(msg.data.team),
+//     };
+// }
```
Comment (mzaks) [inline src/actions/websocket.js]: There is no WebsocketEvents.PATH_TEAM constant present. I commented out the handle function. But it is rather needs to be removed. Comments welcome

File: src/actions/websocket.js [hunk 3]
```diff
@@ -388,7 +400,7 @@ function handleUserAddedEvent(msg) {
         }, getState);
 
         if (msg.broadcast.channel_id === currentChannelId) {
-            dispatch(getChannelStats(teamId, currentChannelId));
+            dispatch(getChannelStats(currentChannelId));
```
Comment (mzaks) [inline src/actions/websocket.js]: based on the function signature, the code before `getChannelStats(teamId, currentChannelId)` was a bug. However I would like to get input from more experienced developers.

File: src/actions/websocket.js [hunk 4]
```diff
@@ -435,7 +447,7 @@ function handleUserRemovedEvent(msg) {
                 EventEmitter.emit(General.SWITCH_TO_DEFAULT_CHANNEL, currentTeamId);
             }
         } else if (msg.data.channel_id === currentChannelId) {
-            dispatch(getChannelStats(currentTeamId, currentChannelId));
+            dispatch(getChannelStats(currentChannelId));
```
Comment (mzaks) [inline src/actions/websocket.js]: See comment on line 403

File: src/actions/websocket.js (hunk 3 above)
Comment (jespino) [reply src/actions/websocket.js]: Yes, that was a bug, clearly, I think is the correct behavior to change it here :)

**Comment by cpanato** [comment]
please ignore the messages from matterbuild. I will check what is wrong when comeback from PTO

File: src/actions/websocket.js [hunk 5]
```diff
@@ -239,9 +253,6 @@ function handleEvent(msg) {
     case WebsocketEvents.UPDATE_TEAM:
         doDispatch(handleUpdateTeamEvent(msg));
         break;
-    case WebsocketEvents.PATCH_TEAM:
```
Comment (reflog) [inline src/actions/websocket.js]: Why was this removed?

Comment (mzaks) [reply src/actions/websocket.js]: I discussed it with @jespino before in a private conversation, sorry I did not write a comment in PR. Long story short, there is no such event 🙂. By introducing Flow, the flow checker showed that `WebsocketEvents.PATCH_TEAM` constant is not existing. I then checked with @jespino and he confirmed that it is not there and there is also no `PATCHED_TEAM` on the BackEnd. So we decided that it would be the best to delete the case and the handle function.

Comment (mzaks) [reply src/actions/websocket.js]: @reflog sorry I am not familiar with the way it is handled correctly, do I get to click on the `Resolve conversation` button? 🙂

Comment (reflog) [reply src/actions/websocket.js]: Thanks for the explanation @mzaks !
About resolve conversation - the originator of the comment presses Resolve if the question was answered (so in this case I will press Resolve) or by the committer if they handled the comment (for example if you fixed the bug mentioned).

File: src/actions/websocket.js (hunk 1 above)
Comment (reflog) [reply src/actions/websocket.js]: Please remove it.

File: src/actions/websocket.js (hunk 2 above)
Comment (reflog) [reply src/actions/websocket.js]: Understood. Please remove it.

Comment (mzaks) [reply src/actions/websocket.js]: All done. The snipped is `Outdated`. All commented out code is already deleted.

File: src/actions/websocket.js (hunk 1 above)
Comment (mzaks) [reply src/actions/websocket.js]: The code is removed. Now resolving the conversation.

**Comment by mzaks** [comment]
Any news on this PR?

**Comment by reflog** [comment]
> Any news on this PR?

@mzaks you forgot to press 'request review' after pushing your changes, so I had no idea there was any work on the PR :)
